common/trinary/trit_array.h

```diff
@@ -0,0 +1,158 @@
+  uint8_t tindex = index % 5U;
+  // Find out the index of the byte in the array
+  index = index / 5U;
+  bytes_to_trits(((byte_t *)trit_array + index), 1, trits, 5);
+  trits[tindex] = trit;
+  trit_array[index] = trits_to_byte(trits, buffer, 5);
+#endif
+}
+
+size_t flex_trit_array_slice(flex_trit_t *to_trit_array,
+                             const flex_trit_t *trit_array, size_t start,
+                             size_t num_trits);
+size_t flex_trit_array_insert(flex_trit_t *trit_array,
+                              const flex_trit_t *from_trit_array, size_t start,
+                              size_t num_trits);
+size_t flex_trit_array_to_int8(trit_t *trits, const flex_trit_t *trit_array,
+                               size_t num_trits);
+size_t int8_to_flex_trit_array(flex_trit_t *trit_array, const trit_t *trits,
+                               size_t num_trits);
+
+typedef struct _trit_array *trit_array_p;
+typedef struct _trit_array {
+  flex_trit_t *trits;
+  size_t num_trits;
+  size_t num_bytes;
+  uint8_t dynamic;
+} trit_array_t;
+
+size_t trit_array_bytes_for_trits(size_t num_trits);
+
+/***********************************************************************************************************
+ * Accessors
+ ***********************************************************************************************************/
+static inline trit_t trit_array_at(trit_array_p trit_array, size_t index) {
+  return flex_trit_array_at(trit_array->trits, index);
+}
+
+static inline void trit_array_set_at(trit_array_p trit_array, size_t index,
+                                     trit_t trit) {
+  flex_trit_array_set_at(trit_array->trits, index, trit);
+}
+
+void trit_array_set_trits(trit_array_p trit_array, flex_trit_t *trits,
+                          size_t num_trits);
+trit_array_p trit_array_slice(trit_array_p trit_array,
+                              trit_array_p to_trit_array, size_t start,
+                              size_t num_trits);
+trit_t *trit_array_to_int8(trit_array_p trit_array, trit_t *trits);
+
+#if !defined(NO_DYNAMIC_ALLOCATION)
+/***********************************************************************************************************
+ * Constructor
+ ***********************************************************************************************************/
+// num_trits is number of trits
+trit_array_p trit_array_new(size_t num_trits);
+
+/***********************************************************************************************************
+ * Destructor
+ ***********************************************************************************************************/
+void trit_array_free(trit_array_p trit_array);
+
+#endif  // NO_DYNAMIC_ALLOCATION
+
+#define TRIT_ARRAY_DECLARE(NAME, NUM_TRITS)                          \
+  size_t NAME##_num_bytes = trit_array_bytes_for_trits(NUM_TRITS);   \
+  flex_trit_t *NAME##_trits[NAME##_num_bytes];                       \
```

Comment:
I think you want to write `flex_trit_t NAME##_trits[NAME##_num_bytes]` here, not create an array of pointers